packages/aws-lambda/src/ssm.js

```diff
@@ -0,0 +1,143 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ * (c) Copyright Instana Inc. and contributors 2022
+ */
+
+'use strict';
+
+const ENV_NAME = 'INSTANA_SSM_PARAM_NAME';
+let fetchedValue;
+let envValue;
+let errorFromAWS;
+let initTimeoutInMs = 0;
+
+module.exports.reset = () => {
+  fetchedValue = null;
+  envValue = null;
+  errorFromAWS = null;
+  initTimeoutInMs = 0;
+};
+
+module.exports.isUsed = () => !!envValue;
+
+module.exports.validate = () => {
+  // NOTE: Always reset the env value, we do not wanne cache, because it might change
```

Comment:
How would the env var change during the lifetime of a Lambda handler instance? I think it can only change when a new Lambda handler with a different environment configuration has been deployed, which will start a new Lambda handler instance, and thereby also restart the underlying container/process. IMO it would be valid to cache the result of this check globally.

Reply:
> I think it can only change when a new Lambda handler with a different environment configuration has been deployed

Yeah I was't sure about that. Usually it works like that - a restart of a the node process is required
My thinking was that it won't hurt to re-read the value (performance wise) if we are not 100% sure

But let me just remove the two lines and cache the value :)